src/Functions/FunctionsStringDistance.cpp

```diff
@@ -108,6 +110,117 @@ struct ByteHammingDistanceImpl
     }
 };
 
+template <bool is_utf8>
+struct ByteJaccardIndexImpl
+{
+    using ResultType = Float64;
+    static ResultType inline process(
+        const char * __restrict haystack, size_t haystack_size, const char * __restrict needle, size_t needle_size)
+    {
+        if (haystack_size == 0 || needle_size == 0)
+            return 0;
+
+        const char * haystack_end = haystack + haystack_size;
+        const char * needle_end = needle + needle_size;
+
+        /// For byte strings use plain array as a set
+        constexpr size_t max_size = std::numeric_limits<unsigned char>::max() + 1;
+        std::array<UInt8, max_size> haystack_set;
+        std::array<UInt8, max_size> needle_set;
```

Comment:
It should have better performance if we replace `std::array` with `std::bitset`.

Reply:
I considered it, but memory benefit is small. In terms of speed we can check if it'll be faster, but I doubt.